client.go

```diff
@@ -386,8 +392,16 @@ func (c *client) periodicQuery(ctx context.Context, params *lookupParams) error
 		if wait == backoff.Stop {
 			return fmt.Errorf("periodicQuery: abort due to timeout")
 		}
+		if timer == nil {
+			timer = time.NewTimer(wait)
+		} else {
+			if !timer.Stop() {
```

Comment:
I don't quite get this case.  Why is it needed?
I get the reasoning for this change, but readability suffers quite some compared to the original logic. Maybe it can be optimized a bit.

Reply:
Yes, this is wrong ;) Timers in Go are a mess.

The only way we can get to this case is when the timer in the `select` statement fired, so there's no reason to stop it. We can just reset it right away.

Reply:
Fixed.